sorting/wiggle_sort.cpp

```diff
@@ -0,0 +1,124 @@
+/**
+ *  \addtogroup sorting Sorting Algorithms
+ *  @{
+ *  \file
+ *  \brief Wiggle Sort Algorithm Implmentation
```

Comment:
Provide a Wikipedia link (if possible) in Markdown format.
If there's no Wikipedia link, you can provide another web source to the algorithm explanation. 🙂